src/app/track/track.component.ts

```diff
@@ -1,346 +1,476 @@
-    const points: Coordinate[] = [];
-    if (this.trackData) {
-      while (i < this.trackData.points.length) {
-        lat = this.trackData.points[i].latitude;
-        long = this.trackData.points[i].longitude;
-        if (lat && long) {
-          const p = fromLonLat([long, lat]);
-          // var p = transform([long, lat], 'EPSG:4326', 'EPSG:3857');
-          points.push(p);
-          trackPoints.push(
-            new Feature({
-              flag: this.trackData.points[i].flag,
-              d1: this.trackData.points[i].d1,
-              d2: this.trackData.points[i].d2,
-              geometry: new Point(p)
-            })
-          );
-          if (this.trackData.points[i].flag === 1) {
-            // L.marker([lat, long]).addTo(trackMapView);
-          }
-          // console.log(this.trackData.points[i].d2);
-        }
-        i++;
-      }
-      trackLines.push(
-        new Feature(
-          new LineString(points)
-        )
-      );
+    ngOnInit() {
+        // Retreive the prefetched track
+        this.route.data.subscribe(
+            (data: { track: Track }) => {
+                this.track = data.track;
+            }
+        );
+        // Load the current user's data
+        this.userService.currentUser.subscribe(
+            (userData: User) => {
+                this.currentUser = userData;
+                this.canModify = (this.currentUser.username === this.track.author.username);
+            });
+        this.populateComments();
+        this.populateTrackData();
     }
 
-    const trackLayer =
-      new VectorLayer({
-        visible: true,
-        updateWhileAnimating: false,
-        updateWhileInteracting: false,
-        source: new VectorSource({
-          features: trackLines
-        }),
-        style: new Style({
-          stroke: new Stroke({
-            width: 3,
-            color: [5, 5, 5, 1]
-          })
+    createMap() {
+        const tileLayer = new TileLayer({
+            source: new OSM(/*{
+      attributions: [
+        'All maps © <a href="https://www.opencyclemap.org/">OpenCycleMap</a>',
+        ATTRIBUTION
+      ],
+      url: 'https://{a-c}.tile.thunderforest.com/cycle/{z}/{x}/{y}.png?apikey=25763459938a4e9d9bec8bde3481f682',
+    }*/)
+        });
+
+
+        const trackPointsD1: Feature<Geometry>[] = [];
+        const trackPointsD2: Feature<Geometry>[] = [];
+        const trackPointsUntaggedD1: Feature<Geometry>[] = [];
+        const trackPointsUntaggedD2: Feature<Geometry>[] = [];
+        const trackLines: Feature[] = [];
+        const points: Coordinate[] = [];
+
+        if (this.trackData) {
+            let filteredPoints = this.trackData.points.filter(point => point.latitude && point.longitude)
+
+            for (let i = 0; i < filteredPoints.length; i++) {
+                let dataPoint = filteredPoints[i];
+                let nextDataPoint = filteredPoints[i + 50] ? filteredPoints[i + 50] : filteredPoints[i];
```

Comment:
I think it would be easier to understand what's going on if the magic value 50 would be stored in a named constant

Reply:
Deleted because it was unused